homeassistant/components/elgato/config_flow.py

```diff
@@ -41,10 +42,12 @@ async def async_step_user(
 
         return self._async_create_entry()
 
-    async def async_step_zeroconf(self, discovery_info: dict[str, Any]) -> FlowResult:
+    async def async_step_zeroconf(
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
+    ) -> FlowResult:
         """Handle zeroconf discovery."""
         self.host = discovery_info[CONF_HOST]
-        self.port = discovery_info[CONF_PORT]
+        self.port = discovery_info[CONF_PORT]  # type: ignore [assignment]
```

Comment:
Type ignore is added here because `ZeroconfServiceInfo` has port defined as `int | None` but `self.port` is set to `int`.
Maybe it is better to adjust `self.port` definition?

Reply:
I think this should be solved by the integration and not by the port. In this case, it can try the default port (I wrote this thing btw...)

```suggestion
        self.port = discovery_info[CONF_PORT] or 9123
```

Reply:
@frenck: done